apps/dgst.c

```diff
@@ -523,28 +524,35 @@ int do_fp(BIO *out, unsigned char *buf, BIO *bp, int sep, int binout,
             BIO_printf(out, "Verified OK\n");
         } else if (i == 0) {
             BIO_printf(out, "Verification Failure\n");
-            return 1;
+            goto end;
         } else {
             BIO_printf(bio_err, "Error Verifying Data\n");
             ERR_print_errors(bio_err);
-            return 1;
+            goto end;
         }
-        return 0;
+        ret = 0;
+        goto end;
     }
     if (key != NULL) {
         EVP_MD_CTX *ctx;
+        int pkey_len;
         BIO_get_md_ctx(bp, &ctx);
-        len = BUFSIZE;
+        pkey_len = EVP_PKEY_size(key);
+        if (pkey_len > BUFSIZE) {
```

Comment:
Do we actually have any signature schemes that exceed this limit? Or have you hit this with an engine supplied signature scheme?

Reply:
This PR is to support our custom engine based signature scheme with large signature sizes. It is originated from #9732. As you commented this issue, PRs are welcome, so I prepared this PR. Hope you'll find it useful. I tried to change existing behavior as little as possible. The new buffer will not be allocated if the existing buffer passed from `dgst_main` is enough to store.

Reply:
No. This is fine as is. I added the labels for the target branches.